frontend/src/api/RestaurantApi.tsx

```diff
@@ -1,17 +1,23 @@
+import { SearchState } from '@/pages/SearchPage';
 import { RestaurantResponse } from '@/types';
 import { useAuth0 } from '@auth0/auth0-react';
 import { useQuery } from 'react-query';
 
 const API_BASE_URL = import.meta.env.VITE_API_BASE_URL;
 
-export const useSearchRestaurants = (city?: string) => {
+export const useSearchRestaurants = (
+  searchState: SearchState,
+  city: string
+) => {
   const { getAccessTokenSilently } = useAuth0();
 
   const createSearchRestaurant = async (): Promise<RestaurantResponse> => {
     const accessToken = await getAccessTokenSilently();
+    const params = new URLSearchParams();
+    params.set('searchQuery', searchState.searchQuery);
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/39#issuecomment-2408112738>#dc9d72</a><span> - 10/11/2024, 09:06 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'searchQuery' parameter is not encoded when added to the URL params, which could lead to potential security vulnerabilities if the search query contains special characters.
</div>
<div id="fix">
<b>Fix:</b> Encode the searchQuery parameter using encodeURIComponent before adding it to the URL params.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -16,7 +16,7 @@
      const accessToken = await getAccessTokenSilently();
      const params = new URLSearchParams();
 -    params.set('searchQuery', searchState.searchQuery);
 +    params.set('searchQuery', encodeURIComponent(searchState.searchQuery));
 
      const response = await fetch(
        `${API_BASE_URL}/api/restaurant/search/${encodeURIComponent(city)}?${params.toString()}`,
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged